### DISCUSSION TITLE
What is Isolated<T> used for?

### DISCUSSION THREAD

**Discussion by iwt-sebastian-boldt** [pointfreeco/isowords#168] [General]
Hey Guys 😉 , 

I stumbled across something I can not fully grasp. 
Inside the `APIClient` Dependency e.g you create an actor to synchronize access to a shared resource.
This actor has some non-isolated properties -> I assume because access to these properties does not need to be isolated because they are declared as `let` anyways.
But then you wrap the Type related to this `nonIsolated` property into something called `Isolated`.
Is there any explanation for this somewhere? I would really appreciate any info about it 🙇 

```swift
nonisolated let baseUrl: Isolated<URL>
```

**Comment by stephencelis** [comment]
Hi @iwt-sebastian-boldt! This type is similar to [our `ActorIsolated` type](https://github.com/pointfreeco/swift-composable-architecture/blob/1c50ffcf440de19e15e3a8b94e7347df0895a6ce/Sources/ComposableArchitecture/Effects/ConcurrencySupport.swift#L297), which we use in the Composable Architecture. It uses a lock under the hood, instead, though, to make a value concurrency-friendly (and sendable) without requiring the overhead of an actor.

We can then use `nonisolated` on the actor to allow for synchronous access without needing to `await`.

We're looking into incorporating this type into our library ecosystem in the future, but wanted to give it some use in isowords first :smile:
